salt/cloud/clouds/proxmox.py

```diff
@@ -740,15 +740,24 @@ def create(vm_):
 def _import_api():
     """
     Download https://<url>/pve-docs/api-viewer/apidoc.js
-    Extract content of pveapi var (json formatted)
+    Extract content of pveapi var (json formatted) if proxmox is version 6
+    Extract content of apiSchema var (json formatted) if proxmox is version 7
     Load this json content into global variable "api"
     """
     global api
     full_url = "https://{}:{}/pve-docs/api-viewer/apidoc.js".format(url, port)
     returned_data = requests.get(full_url, verify=verify_ssl)
 
-    re_filter = re.compile("(?<=pveapi =)(.*)(?=^;)", re.DOTALL | re.MULTILINE)
-    api_json = re_filter.findall(returned_data.text)[0]
+    # Filter below will check for a json variable named either pveapi or apiSchema
+    re_filter = re.compile(
+        "((?<=pveapi =)|(?<=apiSchema =))(.*)(?=^;)", re.DOTALL | re.MULTILINE
+    )
+    # Results for both possible matches are returned as a tuple instead of a single string
+    filter_results = re_filter.findall(returned_data.text)[0]
+    # We need to capture the single tuple item expected to have valid json
+    for result in filter_results:
+        if result != "":
+            api_json = result
```

Comment:
If result == "" then the `api_json` variable would not exist when you make the call to salt.utils.json.loads(api_json) below.

Reply:
"filter_results" is a tuple with the results of both searches. It looks like `('', '{returned json}')` or `('{returned json}', '')`. one of them will always be empty and one of them will always have the expected json unless we've pointed it to an invalid proxmox host or something.

So yes, if `result==""` it should loop once more and catch the expected value. I've run this in both scenarios and the for loop always grabs what is expected and assigns it to `api_json`

let me know if there's anything else here to address to get this merged

Reply:
Coming back to read this again, I think the requested change would be to initiate the `api_json` variable before the for loop and then populate it within the loop so that regardless of what is returned in `filter_results` the `api_json` var always at least exists. I'll get this change made when time allows

Reply:
If `api_json` is `"` then we will still see an exception:

```
>>> api_json = ""
>>> salt.utils.json.loads(api_json)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ch3ll/git/salt/salt/utils/json.py", line 89, in loads
    return json_module.loads(s, **kwargs)
  File "/home/ch3ll/.pyenv/versions/3.8.6/lib/python3.8/json/__init__.py", line 357, in loads
    return _default_decoder.decode(s)
  File "/home/ch3ll/.pyenv/versions/3.8.6/lib/python3.8/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/home/ch3ll/.pyenv/versions/3.8.6/lib/python3.8/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0) 
```

I think the best thing here is if check if `api_json` exists and if it doesn't then return and log an error. Since this was not caught in the tests would you mind adding a test to ensure we log and return an error ?